pkg/solver/che_routing.go

```diff
@@ -280,9 +285,9 @@ func getTrackedEndpointName(endpoint *dw.Endpoint) string {
 // we need to support unique endpoints - so 1 port can actually be accessible
 // multiple times, each time using a different resulting external URL.
 // non-unique endpoints are all represented using a single external URL
-func classifyEndpoints(gatewayEnabled bool, order *int, endpoints *dwo.EndpointList) (singlehostPorts map[int32]map[string]int, multihostPorts map[int32]map[string]int) {
-	singlehostPorts = map[int32]map[string]int{}
-	multihostPorts = map[int32]map[string]int{}
+func classifyEndpoints(gatewayEnabled bool, order *int, endpoints *dwo.EndpointList) (singlehostPorts portMapping, multihostPorts portMapping) {
```

Comment:
I don't like `order` being a pointer. It weirdly splits and complicate simple for loop. Is there any issue if the order number is not continuous? Or could we somehow use return value of the function to determine if we want to increment? Or return current counter and save it to counter in the actual for loop?

I know this PR is not related. It just hit my eyes. I don't insist to fix it here. We can discuss it offline, fix it in another PR, I can fix it...

Reply:
Yep, I would like to have a better (and simple) way of ensuring unique names, too, so let's try to figure that out separately.